lib/kracken/controllers/authenticatable.rb

```diff
@@ -124,6 +112,30 @@ def user_signed_in?
 
       private
 
+      def session_present?
+        session[:user_id] && session[:user_cache_key]
+      end
+
+      def session_and_redis_match?
+        ::SessionManager.get(session[:user_uid]) == session[:user_cache_key]
+      end
+
+      def delete_session_data
+        # Sign out current user
+        session.delete :user_id
+
+        # Clear that user's cache key
+        session.delete :user_cache_key
```

Comment:
I think we're missing the `user_uid` here:

```suggestion
        # Clear that user's cache data
        session.delete :user_uid
        session.delete :user_cache_key
```